hendrix/deploy/ssl.py

```diff
@@ -1,41 +1,53 @@
+from hendrix.facilities.services import HendrixTCPServiceWithTLS
 from .base import HendrixDeploy
-
-from hendrix.contrib import ssl
 from twisted.internet.ssl import PrivateCertificate
 from twisted.protocols.tls import TLSMemoryBIOFactory
+from twisted.internet import reactor
 
 
-class HendrixDeploySSL(HendrixDeploy):
+class HendrixDeployTLS(HendrixDeploy):
     """
     HendrixDeploy encapsulates the necessary information needed to deploy the
     HendrixService on a single or multiple processes.
     """
 
+    def __init__(self, action='start', options={},
+                 reactor=reactor, threadpool=None,
+                 key=None, cert=None, context_factory=None,
+                 context_factory_kwargs=None,
+                 *args, **kwargs):
+
+        super(HendrixDeployTLS, self).__init__(action, options, reactor, threadpool,
+                                               *args, **kwargs)
+        if options.get("https_port") and not options.get("http_port"):
+            self.options["https_only"] = True
+        key = key or self.options['key']
+        cert = cert or self.options['cert']
+        if not key and cert:
```

Comment:
Hmm - Did you intend `not key and not cert` or `not (key and cert)`?

Reply:
I meant `not (key and cert)` - good catch.

Do you prefer `not (key and cert)` or `not key or not cert`?

Reply:
Fixed in my subsequent PR (incoming).